tdigest/accumulation_tree.py

```diff
@@ -0,0 +1,151 @@
+import operator
+from bintrees.rbtree import RBTree, Node
+
+
+class AccumulationTree(RBTree):
+    def __init__(self, mapper, reducer=operator.add, zero=0):
+        RBTree.__init__(self)
+        self._zero = zero
+        self._mapper = mapper
+        self._reducer = reducer
+
+        self._dirty_nodes = set()
+
+        class AccumulatingNode(Node):
```

Comment:
Having this class inside the Tree class is a very clever way to share variables like `_dirty_nodes` 👍